sphinx/ext/napoleon/docstring.py

```diff
@@ -872,6 +884,83 @@ def __init__(self, docstring: Union[str, List[str]], config: SphinxConfig = None
         self._directive_sections = ['.. index::']
         super().__init__(docstring, config, app, what, name, obj, options)
 
+    def _convert_type_spec(self, _type):
```

Comment:
I think more test cases are needed for this function. So it would be better to make this as an independent function.

Reply:
sure, I'll add more tests. It doesn't need `self` anyways